components/commander/common/commander_url.cc

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/commander/common/commander_url.h"
+
+#include <string>
+#include <vector>
+
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_split.h"
+#include "base/strings/string_util.h"
+#include "base/unguessable_token.h"
+
+namespace commander {
+namespace {
+constexpr char kBraveCommandsScheme[] = "brave-commands";
+}  // namespace
+
+GURL GetCommandURL(uint32_t command_index, uint32_t result_set_id) {
+  return GURL(kBraveCommandsScheme + std::string("://") +
+              base::NumberToString(command_index) + "/" +
+              base::NumberToString(result_set_id));
+}
+
+bool TryParseCommandURL(const GURL& url,
+                        uint32_t* command_index,
+                        uint32_t* result_set_id) {
+  if (!url.SchemeIs(kBraveCommandsScheme)) {
```

Comment:
I've added the DCHECK, it might just be me but I feel that passing the args via pointer makes it more obvious that the method is going to change the values. Happy to change to the references if you think they'd be better.

Reply:
No problem! I like pointers for output params :). Mutable references were allowed quite recently, and it feels like it's easy to be misused to me as well.